### PR TITLE
[CI] Add NvFP4 Blackwell Tests

### PR DESCRIPTION
**Pull request by ojhaanshika**
## Purpose

Add unit and integration tests for NvFP4 GEMM. On Blackwell, the relevant Cutlass-based kernels are `cutlass_scaled_fp4_mm` and `flashinfer_scaled_fp4_mm`.

**New file: `tests/kernels/quantization/test_nvfp4_utils.py`**
Unit tests covering:
- `select_nvfp4_linear_backend()` — auto-selection, env var overrides, emulation flag
- `cutlass_fp4_supported()` — sm_100 kernel availability
- `swizzle_blockscale()` — scale layout transformation required by CUTLASS sm_100 kernel
- `pad_nvfp4_weight_for_cutlass()` / `pad_nvfp4_activation_for_cutlass()` — alignment padding for non-standard weight shapes
- `slice_nvfp4_output()` — output trimming after N-dim padding
- `convert_to_nvfp4_linear_kernel_format()` + `apply_nvfp4_linear()` — full weight-prep → GEMM dispatch chain for all backends (VLLM_CUTLASS, FLASHINFER_CUTLASS, FLASHINFER_CUDNN, FLASHINFER_TRTLLM)

**Modified: `tests/models/quantization/test_nvfp4.py`**
Three new integration tests:
- `test_nvfp4_vllm_cutlass` — exercises the vLLM-native CUTLASS sm_100 FP4 kernel (`cutlass_scaled_fp4_mm_sm100a`) end-to-end with `nvidia/Llama-3.1-8B-Instruct-NVFP4`. The pre-existing `test_nvfp4` only covers FlashInfer backends.
- `test_nvfp4_default_backend` — verifies Blackwell auto-selects `FLASHINFER_CUTLASS` when no backend override is set, exercising the `select_nvfp4_linear_backend()` auto-select path end-to-end.
- `test_nvfp4_emulation` — covers the software emulation path (`run_nvfp4_emulations`) via `VLLM_USE_NVFP4_CT_EMULATIONS=1`.

## Test Plan
Unit tests:
python -m pytest tests/kernels/quantization/test_nvfp4_utils.py -v

Integration tests (requires Blackwell + nvidia/Llama-3.1-8B-Instruct-NVFP4):
python -m pytest tests/models/quantization/test_nvfp4.py -v

## Test Result

**Coverage improvement on `nvfp4_utils.py`:**
0% --> 85%

Integration tests: correctness coverage of 3 previously untested code paths (CUTLASS branch, auto-selection, emulation)

### CODE REVIEW

File: tests/kernels/quantization/test_nvfp4_utils.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+@torch.inference_mode()
+def test_swizzle_blockscale_2d(
+    n: int,
+    k: int,
+    device: str,
+) -> None:
+    sf = torch.ones(n, k // BLOCK_SIZE, dtype=torch.float8_e4m3fn, device=device)
+    out = swizzle_blockscale(sf)
+    assert out.dtype == torch.float8_e4m3fn
+    assert out.is_cuda
```
Comment (pavanimajety) [inline tests/kernels/quantization/test_nvfp4_utils.py]: This test seems trivial since we aren't actually testing the swizzle operation itself, could be worth slicing into the out at the indices we expect to change and check that the values are what we expect. Also, would be better to use values other than ones.

File: tests/kernels/quantization/test_nvfp4_utils.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+@pytest.mark.parametrize("device", CUDA_DEVICES)
+@pytest.mark.parametrize("b,n,k", [(2, 128, 64), (4, 256, 128)])
+@torch.inference_mode()
+def test_swizzle_blockscale_3d(
+    b: int,
+    n: int,
+    k: int,
+    device: str,
+) -> None:
+    sf = torch.ones(b, n, k // BLOCK_SIZE, dtype=torch.float8_e4m3fn, device=device)
```
Comment (pavanimajety) [inline tests/kernels/quantization/test_nvfp4_utils.py]: same as above.

**Review by pavanimajety — CHANGES_REQUESTED** [review]
Thanks for the PR, the model tests look good to me. Some feedback for utilities tests.

File: tests/kernels/quantization/test_nvfp4_utils.py (hunk 1 above)
Comment (ojhaanshika) [reply tests/kernels/quantization/test_nvfp4_utils.py]: I've removed the swizzle, padding, and slicing utility tests entirely. These functions are already exercised internally by `test_apply_nvfp4_linear`. Updated in the latest push
